p2p.go

```diff
@@ -603,11 +613,6 @@ func (h *Host) allowSource(src peer.ID) (bool, error) {
 }
 
 // generateKeyPair generates the public key and private key by network address
-func generateKeyPair(masterKey string) (crypto.PrivKey, crypto.PubKey, error) {
-	hash := sha1.Sum([]byte(masterKey))
-	seedBytes := hash[12:]
-	seedBytes[0] = 0
-	seed := int64(binary.BigEndian.Uint64(seedBytes))
-	r := rand.New(rand.NewSource(seed))
-	return crypto.GenerateKeyPairWithReader(crypto.Ed25519, 2048, r)
```

Comment:
Is there any indication from documentation the relevance of this master key value? It may prove useful to have these keys generated from a master key.

Reply:
@danielcooperxyz  There's no documentation for it, if there's no master provided it just uses a string of the ip and port of the network as a seed so I'm not too sure. 

The linter complained about `math/rand` not being secure so I changed it to the way we create keys within Huski